fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -11,28 +11,52 @@
 namespace fml {
 
 MessageLoopFuchsia::MessageLoopFuchsia()
-    : loop_(&kAsyncLoopConfigAttachToCurrentThread) {}
+  : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+    running_(false) {
+  auto status = zx_timer_create(ZX_TIMER_SLACK_LATE, ZX_CLOCK_MONOTONIC, &timer_);
+  FML_CHECK(status == ZX_OK);
+  timer_wait_ = new async::Wait(timer_, ZX_TIMER_SIGNALED);
+
+  auto wait_handler = [&](async_dispatcher_t* dispatcher,
+                        async::Wait* wait,
+                        zx_status_t status,
+                        const zx_packet_signal_t* signal
```

Comment:
Maybe a DCHECK to ensure that signal is `ZX_TIMER_SIGNALED`

Reply:
I made everything FML_CHECK as per @iskakaushik's request offline